subsys/bluetooth/host/hci_core.c

```diff
@@ -3808,109 +4408,307 @@ void bt_data_parse(struct net_buf_simple *ad,
+		listener->timeout();
+	}
 }
 
-#if defined(CONFIG_BT_HCI_VS_EVT_USER)
-int bt_hci_register_vnd_evt_cb(bt_hci_vnd_evt_cb_t cb)
+/* Convert Extended adv report evt_type field into adv type */
+static u8_t get_adv_type(u8_t evt_type)
 {
-	hci_vnd_evt_cb = cb;
-	return 0;
+	switch (evt_type) {
+	case 19: return BT_GAP_ADV_TYPE_ADV_IND;
+	case 21: return BT_GAP_ADV_TYPE_ADV_DIRECT_IND;
+	case 18: return BT_GAP_ADV_TYPE_ADV_SCAN_IND;
+	case 16: return BT_GAP_ADV_TYPE_ADV_NONCONN_IND;
+	case 26: /* Scan response from non-connectable advertiser */
+	case 27: return BT_GAP_ADV_TYPE_SCAN_RSP;
+	default: return 0;
+	}
 }
-#endif /* CONFIG_BT_HCI_VS_EVT_USER */
 
-static void hci_vendor_event(struct net_buf *buf)
+static void le_adv_ext_report(struct net_buf *buf)
 {
-	bool handled = false;
+	u8_t num_reports = net_buf_pull_u8(buf);
 
-#if defined(CONFIG_BT_HCI_VS_EVT_USER)
-	if (hci_vnd_evt_cb) {
-		struct net_buf_simple_state state;
+	BT_DBG("Adv number of reports %u",  num_reports);
 
-		net_buf_simple_save(&buf->b, &state);
+	while (num_reports--) {
+		struct bt_hci_evt_le_ext_advertising_info *evt;
+		struct bt_le_scan_recv_info adv_info;
+
+		if (buf->len < sizeof(*evt)) {
+			BT_ERR("Unexpected end of buffer");
+			break;
+		}
+
+		evt = net_buf_pull_mem(buf, sizeof(*evt));
+
+		adv_info.primary_phy = evt->prim_phy;
+		adv_info.secondary_phy = evt->sec_phy;
+		adv_info.tx_power = evt->tx_power;
+		adv_info.rssi = evt->rssi;
+		adv_info.sid = evt->sid;
+
+		adv_info.adv_type = get_adv_type(evt->evt_type);
+		adv_info.adv_props = evt->evt_type;
+
+		le_adv_recv(&evt->addr, &adv_info, buf, evt->length);
+	}
+}
+#endif /* defined(CONFIG_BT_ADV_EXT) */
+
+static void le_adv_report(struct net_buf *buf)
+{
+	u8_t num_reports = net_buf_pull_u8(buf);
+	struct bt_hci_evt_le_advertising_info *evt;
+
+	BT_DBG("Adv number of reports %u",  num_reports);
+
+	while (num_reports--) {
+		struct bt_le_scan_recv_info adv_info;
+
+		if (buf->len < sizeof(*evt)) {
+			BT_ERR("Unexpected end of buffer");
+			break;
+		}
+
+		evt = net_buf_pull_mem(buf, sizeof(*evt));
+
+		adv_info.rssi = evt->data[evt->length];
+		adv_info.primary_phy = BT_GAP_LE_PHY_1M;
+		adv_info.secondary_phy = 0;
+		adv_info.tx_power = BT_GAP_TX_POWER_INVALID;
+		adv_info.sid = BT_GAP_SID_INVALID;
+
+		adv_info.adv_type = evt->evt_type;
+		adv_info.adv_props = get_adv_props(evt->evt_type);
+
+		le_adv_recv(&evt->addr, &adv_info, buf, evt->length);
+
+		net_buf_pull(buf, evt->length + sizeof(adv_info.rssi));
+	}
+}
+#endif /* CONFIG_BT_OBSERVER */
+
+#if defined(CONFIG_BT_ADV_EXT)
+#if defined(CONFIG_BT_BROADCASTER)
+static void le_adv_set_terminated(struct net_buf *buf)
+{
+	struct bt_hci_evt_le_adv_set_terminated *evt;
+	struct bt_le_ext_adv *adv;
+
+	evt = (void *)buf->data;
+	adv = bt_adv_lookup_handle(evt->adv_handle);
+
+	BT_DBG("status 0x%02x adv_handle %u conn_handle 0x%02x num %u",
+	       evt->status, evt->adv_handle, evt->conn_handle,
+	       evt->num_completed_ext_adv_evts);
+
```

Comment:
Good point. It is actually not completely added yet. So will fix that as well.

Reply:
@Rasmus-Abildgren-Bose I have completed the handling of this now. Does it still need a comment you think? I added some helper functions that makes it clearer IMO.

Reply:
:) I, in general, think there are very little comments in the code, which I think is a shame, but since it is the style, I can live with it. I do want to state that it is much easier to comment while developing than bugfixing later